clients/drcachesim/analyzer.h

```diff
@@ -115,30 +116,59 @@ template <typename RecordType, typename ReaderType> class analyzer_tmpl_t {
 protected:
     typedef scheduler_tmpl_t<RecordType, ReaderType> sched_type_t;
 
+    // Tool data for one shard.
+    struct analyzer_tool_shard_data_t {
+        analyzer_tool_shard_data_t()
+            : shard_data_(nullptr)
+        {
+        }
+        analyzer_tool_shard_data_t(analyzer_tool_shard_data_t &&src)
+        {
+            shard_data_ = src.shard_data_;
+            interval_snapshot_data_ = std::move(interval_snapshot_data_);
+        }
+
+        void *shard_data_;
+        std::queue<typename analysis_tool_tmpl_t<RecordType>::interval_state_snapshot_t *>
+            interval_snapshot_data_;
+    };
+
+    // Data for one trace shard.
+    struct analyzer_shard_data_t {
+        analyzer_shard_data_t()
+            : cur_interval_index_(0)
+        {
+        }
+        analyzer_shard_data_t(analyzer_shard_data_t &&src)
+        {
+            cur_interval_index_ = src.cur_interval_index_;
+            tool_data_ = std::move(tool_data_);
+        }
+
+        uint64_t cur_interval_index_;
+        std::vector<analyzer_tool_shard_data_t> tool_data_;
+    };
+
     // Data for one worker thread.  Our concurrency model has each input shard
     // analyzed by a single worker thread, eliminating the need for locks.
     struct analyzer_worker_data_t {
         analyzer_worker_data_t(int index, typename sched_type_t::stream_t *stream)
-            : index(index)
-            , stream(stream)
-            , cur_interval_index(0)
-            , shard_data()
+            : index_(index)
+            , stream_(stream)
         {
         }
         analyzer_worker_data_t(analyzer_worker_data_t &&src)
         {
-            index = src.index;
-            stream = src.stream;
-            cur_interval_index = src.cur_interval_index;
-            shard_data = std::move(src.shard_data);
-            error = std::move(src.error);
+            index_ = src.index_;
+            stream_ = src.stream_;
+            shard_data_ = std::move(src.shard_data_);
+            error_ = std::move(src.error_);
         }
 
-        int index;
-        typename scheduler_tmpl_t<RecordType, ReaderType>::stream_t *stream;
-        std::string error;
-        uint64_t cur_interval_index;
-        std::unordered_map<int, std::vector<void *>> shard_data;
+        int index_;
```

Comment:
style: We use trailing underscores only for class fields

Reply:
Done